stack/loader.go

```diff
@@ -103,25 +103,23 @@ func (l Loader) Set(dir string, s S) {
 	l.stacks[dir] = s
 }
 
-// LoadAll loads all the stacks in the dirs directories. If dirs are relative
-// paths, then basedir is used as base.
-func (l Loader) LoadAll(root string, basedir string, dirs ...string) ([]S, error) {
+// LoadAll loads all the stacks in the dirs directories. If dirs are project
+// paths, then absBaseDir is used as base.
+func (l Loader) LoadAll(root string, absBaseDir string, dirs ...string) ([]S, error) {
```

Comment:
I imagine there is good reason for this and Im missing something, but can't we work just with root + dirs is always projRelPaths ? or paths relative to root ? It is considerably complicated to my limited CPU that we have root, which a lot of stuff is relative to, and then another absBaseDir..that maybe things are going to be relative to ? and then paths...that may or may not be relative to absBaseDir ? (docs mention "if", so I imagine it is not always relative to absBaseDir). For example, Im wondering something like:

1 - "dirs" could be absolute ? (not very clear from docs)
2 - "dirs" could be relative to absBaseDir ? (clear from docs)
3 - "dirs" could be relative to root ? (not clear from docs)

Maybe when stacks are created/loaded they should normalize their after/before/wants in a single place so it is always project abs path relative to root (like /etc/etc), so any further logic can always count on that, or else this complexity (having /path/1 and ../../path/1, etc) spreads around, like here. I could be way off since you have been working more on this than me...Im just sharing that I got confused, but I do get confused rather easily :laughing:

Reply:
This is unrelated to the PR itself, just sharing the confusion/feelings.

Reply:
It is kinda like we have 2 ways to represent the same thing and we are allowing that to spread, maybe in this case we can allow users to have 2 ways to reference the same thing but then internally we normalize to a single one. Just sharing some thoughts.

Reply:
I improved the docs, let me know if it makes more sense now.